engine/base/runtime/src/main/scala/pl/touk/nussknacker/engine/baseengine/metrics/dropwizard/BaseEngineMetrics.scala

```diff
@@ -0,0 +1,37 @@
+package pl.touk.nussknacker.engine.baseengine.metrics.dropwizard
+
+import com.typesafe.config.Config
+import com.typesafe.scalalogging.LazyLogging
+import io.dropwizard.metrics5.MetricRegistry
+import pl.touk.nussknacker.engine.baseengine.metrics.dropwizard.influxdb.BaseEngineInfluxDbReporter
+import pl.touk.nussknacker.engine.util.loader.ScalaServiceLoader
+
+import scala.util.control.NonFatal
+
+object BaseEngineMetrics extends LazyLogging {
+
+  def prepareRegistry(config: Config): MetricRegistry = {
+    val metricRegistry = new MetricRegistry
+    val metricReporters = loadMetricsReporters()
+    if (metricReporters.nonEmpty) {
+      metricReporters.foreach { reporter =>
+        reporter.createAndRunReporter(metricRegistry, config)
+      }
+    } else {
+      BaseEngineInfluxDbReporter.createAndRunReporterIfConfigured(metricRegistry, config)
+    }
+    metricRegistry
+  }
+
+  private def loadMetricsReporters(): List[MetricsReporter] = {
+    try {
+      val reporters = ScalaServiceLoader.load[MetricsReporter](Thread.currentThread().getContextClassLoader)
```

Comment:
Why it was needed to by loaded via ServiceLoader? (just asking)

Reply:
TBH I just copied it from StandaloneApp. The idea was to be able to add other providers with custom jars, not sure if it was much used, can change it, but how would you like it to work?